pre_commit/languages/docker.py

```diff
@@ -15,6 +18,32 @@
 healthy = helpers.basic_healthy
 
 
+def is_in_docker() -> bool:
+    try:
+        with open('/proc/1/cgroup') as f:
+            return 'docker' in f.read()
+    except FileNotFoundError:
+        return False
+
+
+def get_docker_path(path: str) -> str:
+    if not is_in_docker():
+        return path
+    hostname = socket.gethostname()
+    docker_output = json.loads(
+        subprocess.check_output(['docker', 'inspect', hostname]),
+    )
+    for mount_path in docker_output[0]['HostConfig']['Binds']:
+        src_path, to_path = mount_path.split(':')
+        if to_path == path:
+            return src_path
+        elif path.startswith(to_path):
+            return path.replace(to_path, src_path)
```

Comment:
this is slightly error prone, you could have `/foo/bar` and `/foo/barbutsomethingelse` and it would do the wrong thing --  I think you have to use the special path functions to figure this out (I believe there's a prefix function somewhere in `os.path`)

Reply:
done